### PR TITLE
Sketching blog

### PR DESCRIPTION
**Pull request by Kryan92**

### CODE REVIEW

File: _posts/2017-04-13-sketching-in-code.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+title: "Sketching in code to build online services"
+category: [blog]
+tags: 'technology'
+author: Simon Schwartz
+author-excerpt: "Simon Schwartz is a frontend developer at the Digital Transformation Agency."
+thumbnail: /images/blog-thumbnails/Sketch_in_code_sml.png
+hero-image: /images/blog-content/Sketch_in_code_hero.png
+---
+
+!Four developers working together in an office.]({{ site.url }}{{ site.baseurl }}{{ page.hero-image }})
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: Needs a [ after the ! e.g. change to 
`Four developers working together in an office.`

File: _posts/2017-04-13-sketching-in-code.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+*Caption: DTA developers at work.*
+
+A great online service is the result of bringing together excellent design and code. To deliver these services it is critical designers and developers work together. 
+
+It is no use if a designer gives a developer a set of high fidelity specs that don’t consider the constraints of the technology the product is being built in, or the browsers it will be displayed on.
+
+Luckily, this is a pretty easy problem to solve—allow designers and developer to work closely together, and sketch in code. At the DTA, this approach is BAU, and produces great results.
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: We should define BAU. Change it to 
`<abbr title="Business as usual">BAU</abbr>`

File: _posts/2017-04-13-sketching-in-code.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+## Ideas are cheap, actually making stuff is where you begin to win the battle
+
+It is good practice to test your ideas early. By doing this you are able to identify some of the constraints you may face—whether it is navigation, layout, accessibility or mobile responsiveness. The best way to do this is by sketching in code, prototyping your work early and iterating quickly.
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: Content guide says to use [spaced em-dashes](http://content-guide.apps.staging.digital.gov.au/punctuation-grammar/#em-dashes-mdash)

File: _posts/2017-04-13-sketching-in-code.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+The outcome of prototyping is to:
+
+- define a clear and shared understanding of what you are going to build in beta
+- confirm your original idea or hypothesis is valid
+
+It is important to remember that prototyping shouldn’t be the same as building production software. It’s about rapidly testing and validating ideas before committing to building production software.
+
+I recommend the following guidelines for prototyping.
+
+## You don't need to build it right, until you know you are building the right thing
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: I think this should be a h3

File: _posts/2017-04-13-sketching-in-code.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+In most cases it is more valuable to have a rough, low fidelity prototype that covers the entire user journey you are testing. Testing well built, high fidelity features in isolation won’t give you as many meaningful user insights during usability testing. 
+
+## Don’t waste time on code quality
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: I think this should be a h3

File: _posts/2017-04-13-sketching-in-code.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+Depending on what you need to test with your prototype, it may be viable to omit features such as:
+
+- cross browser/device testing
+- unit and integration tests
+- high fidelity visual design
+- accessibility
+
+Commit messages and pull requests could also be brief and merely name the piece of work being done.
+
+## Keep the tech simple
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: I think this should be a h3

File: _posts/2017-04-13-sketching-in-code.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+Evaluate how you want to build the prototype. Remember that the most important thing is to be able to build ideas out quickly. Some ideas you build will get disproved and you will have to delete them anyway. For static prototypes we use Jekyll, a static site generator. For interactive prototypes that need to persist data such as decision tools or forms, we use React. The key is to choose a technology that makes it quick and easy to build and iterate the team’s ideas.
+
+## Fake it if you can
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: I think this should be a h3

File: _posts/2017-04-13-sketching-in-code.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+Evaluate whether you need to integrate with existing services or real data. It may be simpler to fake interactions with JavaScript than to integrate with production systems. You need to decide whether it is viable to test your ideas without using real data.
+
+## Make it easy for everyone to contribute
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: I think this should be a h3

File: _posts/2017-04-13-sketching-in-code.md (hunk 7 above)
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: Add a link to https://jekyllrb.com/ and https://facebook.github.io/react/

File: _posts/2017-04-13-sketching-in-code.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+Try to make it easy for non-technical people in the team to contribute to the prototype. While this isn’t critical it’s always better to harness the collective intelligence of the team. We should remove the developers as blockers for ideas getting into the prototype. This may be showing your team mates how to edit markdown files in GitHub, or editing a JSON blob consumed by your prototype.
+
+## Want to learn more?
+
+DTA Service design handbook - [http://ausdto.github.io/service-handbook/alpha/](http://ausdto.github.io/service-handbook/alpha/)
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: It's best if we make these descriptive links

File: _posts/2017-04-13-sketching-in-code.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+Jalpha, a rubygem for scaffolding out new opinionated Jekyll projects - https://github.com/AusDTO/jalpha
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: It's best if we make these descriptive links

File: _posts/2017-04-13-sketching-in-code.md (hunk 1 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 2 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 3 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 4 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Me too

File: _posts/2017-04-13-sketching-in-code.md (hunk 5 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 6 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 8 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 7 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 6 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 9 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 10 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md (hunk 8 above)
Comment (Kryan92) [inline _posts/2017-04-13-sketching-in-code.md]: Done

File: _posts/2017-04-13-sketching-in-code.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+It is no use if a designer gives a developer a set of high fidelity specs that don’t consider the constraints of the technology the product is being built in, or the browsers it will be displayed on.
+
+Luckily, this is a pretty easy problem to solve—allow designers and developer to work closely together, and sketch in code. At the DTA, this approach is business as usual, and produces great results.
+
+## Ideas are cheap, actually making stuff is where you begin to win the battle
+
+It is good practice to test your ideas early. By doing this you are able to identify some of the constraints you may face — whether it is navigation, layout, accessibility or mobile responsiveness. The best way to do this is by sketching in code, prototyping your work early and iterating quickly.
+
+When developing a product you need to sketch out what it will look like—prototype it. Prototyping takes an idea or hypothesis and demonstrates how you plan to address it. It makes sure that your idea (or your team’s idea) is valid. It makes it easy for others to have a shared understanding of what you are trying to do.
```
Comment (bjgordon) [inline _posts/2017-04-13-sketching-in-code.md]: Sorry another em-dash to space :)

**Pull request merged**